File: src/app/models/Cell.ts
export default class Cell {
  orientations: Array<Orientation>
  block_group: BlockGroup


  constructor(orientations: Array<Orientation>, block_group: BlockGroup) {
    this.orientations = orientations;
    this.block_group = block_group;
  }

}

export enum BlockGroup {
  VOID = "VOID",
  WALL = "WALL",
  DOT = "DOT",
  SUPERDOT = "SUPERDOT",
  OBJECT = "OBJECT",
}

export enum Orientation {
  LEFT = "LEFT",
  RIGHT = "RIGHT",
  BOTTOM = "BOTTOM",
  TOP = "TOP",
}

File: src-tauri/src/models/cell.rs
use serde::{Deserialize, Serialize};

#[derive(Deserialize, Serialize, Clone)]
pub struct Cell {
    pub orientations: Vec<Orientation>,
    pub block_group: BlockGroup,
}

#[derive(Deserialize, Serialize, Clone)]
pub enum BlockGroup {
    VOID,
    WALL,
    DOT,
    SUPERDOT,
    OBJECT,
}

#[derive(Deserialize, Serialize, Clone)]
pub enum Orientation {
    LEFT,
    RIGHT,
    BOTTOM,
    TOP,
}